server/routes/regionRouter.js

```diff
@@ -0,0 +1,69 @@
+const express = require('express');
+const helper = require('./utils');
+const Session = require('../models/Session');
+const RegionService = require('../services/RegionService');
+
+const regionRouter = express.Router();
+
+regionRouter.get(
+  '/',
+  //   helper.apiKeyHandler,
+  helper.errorHandler,
+  helper.handlerWrapper(async (req, res, next) => {
+    const { ownerId } = req.query;
+    const session = new Session();
+    const regionSerivce = new RegionService(session);
+    const ownerRegions =
+      regionSerivce.regionRepository.getAllByOwnerId(ownerId);
+    res.status(200).json({
+      regions: ownerRegions,
+    });
+  }),
+);
+
+regionRouter.get(
+  '/:regionId',
+  //   helper.apiKeyHandler,
+  helper.errorHandler,
+  helper.handlerWrapper(async (req, res, next) => {
+    const { regionId } = req.params;
+    const session = new Session();
+    const regionSerivce = new RegionService(session);
+    const region = regionSerivce.getById(regionId);
+    res.status(200).json({
+      region,
+    });
+  }),
+);
+
+regionRouter.post(
+  '/',
+  //   helper.apiKeyHandler,
+  helper.errorHandler,
+  helper.handlerWrapper(async (req, res, next) => {
+    const region = req.body;
+    const session = new Session();
+    const regionSerivce = new RegionService(session);
+    const newRegion = await regionSerivce.createRegion(region);
+    res.status(200).json({
+      region: newRegion,
+    });
+  }),
+);
```

Comment:
You need errorHandler here? I remember if you use `handleWrapper` to wrap the handler function, then all is fine, if there are any error throw from the internal call stacks, then the error will be catched by the `errorHandler` because we set it here:


https://github.com/bstetzer32/treetracker-regions-api/blob/d55901130e98c0621151d52169213ab1fdeeb0c0/server/app.js#L40-L41

Reply:
you can do an experiment, by throwing any error in the handler, if the client receive error like 500, then errorhandler is working.

The purpose of it is that: if we don't hanler the error then the client will be blocked and waiting endless.

Reply:
You are correct, I tested and the error is handled properly and sent back to the client. Removed from branch